### PR TITLE
Update publish action to remove deprecated publish-nuget steps

### PR DESCRIPTION
**Pull request by paulmilbank**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,61 +1,99 @@
 # https://github.com/marketplace/actions/publish-nuget
 name: publish to nuget
 on:
+  workflow_dispatch:
```
Comment (mcuzmenco) [inline .github/workflows/publish.yml:4]: This allows manual trigger?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -1,61 +1,99 @@
...
-          PROJECT_FILE_PATH: source/SemVerAnalyzer/SemVerAnalyzer.csproj
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # Get all history to allow automatic versioning using MinVer
 
-          # NuGet package id, used for version detection & defaults to project name
-          # PACKAGE_NAME: Core
+    # Install the .NET SDK indicated in the global.json file
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v4
```
Comment (mcuzmenco) [inline .github/workflows/publish.yml:31]: Don't we need to specify `dotnet-version`?

**Review by mcuzmenco — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (paulmilbank) [reply .github/workflows/publish.yml:4]: Yes, I believe so

File: .github/workflows/publish.yml (hunk 2 above)
Comment (paulmilbank) [reply .github/workflows/publish.yml:31]: Looks like if we have a global.json with the SDK in it we don't need to specify it in the yml files

**Pull request merged**